src/osd/osd_types.h

```diff
@@ -1380,7 +1380,16 @@ struct pg_pool_t {
   std::map<std::string, std::string> properties;  ///< OBSOLETE
   std::string erasure_code_profile; ///< name of the erasure code profile in OSDMap
   epoch_t last_change = 0;      ///< most recent epoch changed, exclusing snapshot changes
-
+  // If non-zero, require OSDs in at least this many different instances...
+  uint32_t peering_crush_bucket_count = 0;
+  // of this bucket type...
+  uint32_t peering_crush_bucket_barrier = 0;
+  // including this one
+  int32_t peering_crush_mandatory_member = 0;
```

Comment:
I may have missed something, but for this to be effective, PastIntervals::check_new_interval needs to consider this when dealing with maybe_went_rw.

Reply:
Same with bucket_count/barrier.